ark/schema/shared/utils.ts

```diff
@@ -61,27 +61,49 @@ export type PathToPropStringFn = <stringifiable>(
 	:	NoInfer<[opts: PathToPropStringOptions<stringifiable>]>
 ) => string
 
-export const pathToPropString: PathToPropStringFn = (path, ...[opts]) => {
+export type AppendPropToPathStringFn = <stringifiable>(
+	path: string,
+	prop: stringifiable,
+	...[opts]: [stringifiable] extends [PropertyKey] ?
+		[opts?: PathToPropStringOptions]
+	:	NoInfer<[opts: PathToPropStringOptions<stringifiable>]>
+) => string
+
+export const appendPropToPathString: AppendPropToPathStringFn = (
+	path,
+	prop,
+	...[opts]
```

Comment:
Can this just be `opts` without TS complaining? The signature there is really just a TS trick to force opts to be passed in some situations.

Reply:
No it cannot
It's possible to make it a `function` with overloads and loose implementation though

Reply:
Actually I can make it required here

Reply:
No that didn't work